cfpb/sbl-frontend
# [Prettier] get linting going for vscode

Fixes Prettier in VSCode

Closed #150 

## Changes
- change `"prettier"` to `"plugin:prettier/recommended"` as per the `eslint-plugin-prettier` [set up instructions](https://github.com/prettier/eslint-plugin-prettier)
- change the default formatter to eslint
- run `yarn dlx @yarnpkg/sdks vscode` to [get the sdks](https://yarnpkg.com/getting-started/editor-sdks) needed to run prettier
- Disable `"unicorn/no-nested-ternary": ["error"],` because it [conflicts with eslint](https://github.com/prettier/eslint-config-prettier/issues/110)
<img width="1153" alt="Screenshot 2024-01-07 at 11 08 55 AM" src="https://github.com/cfpb/sbl-frontend/assets/19983248/7ce158e7-d855-4bfa-b4a8-ac273c091759">

## How to test this PR

1. Restart VSCode after checking out this branch
2. Go to a `.ts/.tsx` file
3. Change the indentation unnecessarily
4. Save
5. Should see the indentation correct itself

## Screenshots

![Screenshot 2024-01-08 at 10 11 27 AM](https://github.com/cfpb/sbl-frontend/assets/19983248/a461b531-9863-4223-89d0-7a819535e0b2)


## Notes

- All the sdk file changes are auto-generated by the command `yarn dlx @yarnpkg/sdks vscode`
